### PR TITLE
[WIP] Lid-driven cavity verification experiment (and example)

### PR DESCRIPTION
**Pull request by ali-ramadhan**
This PR adds a work-in-progress lid-driven cavity verification experiment. It also somewhat acts as a test for PR #475.

It's a very classic test case and makes for a nice simple example: https://www.cfd-online.com/Wiki/Lid-driven_cavity_problem

I will compare point-wise velocity values along vertical and horizontal lines through the geometric center of the cavity at a resolution of 128x128 for various Reynolds numbers from Re=100 to Re=10,000. The values are reported in Tables 1 and 2 of [Ghia et al. (1982)](https://doi.org/10.1016/0021-9991(82)90058-4).

Basically doing what this YouTube video is doing: https://www.youtube.com/watch?v=gkg3nG_qU7g

Helps with #332 and #346

### CODE REVIEW

**Comment by ali-ramadhan** [comment]
Re=400 test case at 64x64 looks qualitatively okay. Will revisit over the next few days to do a detailed verification.

image

**Comment by ali-ramadhan** [comment]
This branch's history got weird. Will reopen a fresh branch.
